javase/src/main/java/com/google/zxing/client/j2se/ImageReader.java

```diff
@@ -47,7 +50,13 @@ public static BufferedImage readImage(URI uri) throws IOException {
       throw new IOException("Resource not found: " + uri, iae);
     }
     if (result == null) {
-      throw new IOException("Could not load " + uri);
+      Opener opener_ = new Opener();
```

Comment:
For the TIFF format, I have read a few forum and they say that you need another plugin (http://stackoverflow.com/questions/1954685/cant-read-and-write-a-tiff-image-file-using-java-imageio-standard-library). I am trying to read some 1-page scans (I am not allowed to show them). The code is only trying to read the first page (maybe as a future modification, the choice can be given to the user). I have looked at JAI but as you said, it is really old and by using ImageJ, a lot of image processing tools are imported (therefore they can be used in order to improve the detection).
For maven, it is my first time using it, I did not realize that you have a repository. I found ImageJ on it, I will add this dependency (and it will solve the other comment). Thank you for your comments.
For the license, the only thing that I found is on wikipedia: http://imagej.net/Licensing (it seems that the license is unclear due to the participation of non us government employee to a government code but it is in the public domain).
I do not know if the code is bloated (I did not looked at the source code) but the zip that you can download on their website is about 60MB.